### DISCUSSION TITLE
Client Code vs Server Code Clarification

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#13959] [Help]
Newbie trying to get some clarification around pages and what part of the page is client vs server and is that consistent.

I will make an assumption that getServerSideProps, getStaticProps, and getStaticPaths are server-side, but is the rest of the component client-side or only client-side in a React hook.  And does this change if we set fallback=true?

Is this identified in a document?

Reason I ask is if I follow the sample in this for redirects: https://github.com/vercel/next.js/discussions/11346, it works.  But if the Component is SSG with fallback=true, the useEffect does not work the same and many times I get errors related to client headers have already been written.

So based on the question above, is everything in the component server-side except for hooks, and then does that change if SSG is set for fallback=true.  And one more question, does the server vs client change if its set to fallback=true but you hit a page that doesn't exist (which would normally be a 404)?

**Comment by Timer** [comment]
> I get errors related to client headers have already been written.

This was a bug fixed on `next@canary`. Please upgrade!

---

To view exactly what's on the server vs client, put your code into this tool:
https://next-code-elimination.now.sh/

**Reply by jaybytez** [reply]
@Timer, thanks for the follow-up.

Just for the clarification, is everything in a page server-side except for what is in a hook, which then should be client-side rendered?  And regardless of SSG or SSR?
